glib/gsocket.go

```diff
@@ -0,0 +1,43 @@
+package glib
+
+// #include <gio/gio.h>
+import "C"
+import (
+	"errors"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/exp/constraints"
+)
+
+type Socket struct {
+	*Object
+}
+
+func SocketNew(domain, typ, proto int) (*Socket, error) {
+	fd, err := syscall.Socket(domain, typ, proto)
+	if err != nil {
+		return nil, err
+	}
+	return SocketNewFromFd(fd)
+}
+
+func SocketNewFromFd[T constraints.Integer](fd T) (*Socket, error) {
```

Comment:
Okay I get it now. The windows syscall returns a `Handle` https://pkg.go.dev/syscall@go1.22.2?GOOS=windows#Socket and the linux one an int https://pkg.go.dev/syscall@go1.22.2?GOOS=linux#Socket

Reply:
commentary about generic added